common/utils/src/main/scala/org/apache/spark/internal/LogKey.scala

```diff
@@ -16,10 +16,14 @@
  */
 package org.apache.spark.internal
 
+import java.util.Locale
+
 /**
  * All structured logging `keys` used in `MDC` must be extends `LogKey`
  */
-trait LogKey
+trait LogKey {
+  val name: String = this.toString.toLowerCase(Locale.ROOT)
```

Comment:
We can also make it `lazy val` here. I am not sure if the cost will be eventually higher. I suggest keeping it simple for now.